tests/test_star_rating.py

```diff
@@ -0,0 +1,376 @@
+        """,
+    },
+    {
+        "expected": 5,
+        "html": """
+        <span class="rating ">
+          <span style="width: 100%"></span>
+        </span>
+
+        """,
+    },
+    {
+        "expected": 3,
+        "html": """
+        <span class="star-rating__gold review-rating-star" style="width:60%"></span>
+        """,
+    },
+    {
+        "expected": 4,
+        "html": """
+        <img src="data:image/svg+xml;base64,..." alt="star" title="4">
+        """,
+    },
+    {
+        "expected": 5,
+        "html": """
+        <span class="stars-container" tabindex="" aria-label="5.0 Stars. 0 reviews.">
+        """,
+    },
+    {
+        "expected": 3,
+        "html": """
+        <div class="TTratingBox TTrating-3-0">
+        """,
+    },
+    {
+        "expected": 5,
+        "html": """
+        <div role="img" class="ebay-star-rating">
+          <span class="star-rating">
+            <i class="fullStar"></i>
+            <i class="fullStar"></i>
+            <i class="fullStar"></i>
+            <i class="fullStar"></i>
+            <i class="fullStar"></i>
+          </span>
+        </div>
+        """,
+    },
+    {
+        "expected": 5,
+        "html": """
+        <div class="stars stars50" style="height:initial"><span class="f24 fc3" style="padding-left:10px"> </span></div>
+        """,
+    },
+    {
+        "expected": 4,
+        "html": """
+        <div class="rating" style="width:80%">
+        <div class="relative whitespace-no-wrap inline-block text-sm" style="width:95px; height:21px;">
+        <div class="absolute">
+          <i class="fa fa-star text-primary-light-v1"></i><i class="fa fa-star text-primary-light-v1"></i><i class="fa fa-star text-primary-light-v1"></i>
+          <i class="fa fa-star text-primary-light-v1"></i><i class="fa fa-star text-primary-light-v1"></i>
+        </div>
+        <div class="absolute overflow-hidden" style="width:80%">
+          <i class="fa fa-star text-primary-v1"></i><i class="fa fa-star text-primary-v1"></i><i class="fa fa-star text-primary-v1"></i>
+          <i class="fa fa-star text-primary-v1"></i><i class="fa fa-star text-primary-v1"></i>
+        </div>
+        </div>
+        """,
+    },
+    {
+        "expected": 1,
+        "xfail": True,
+        "html": """
+        <div aria-hidden="true" class="pr-rating-stars">
+          <div class="pr-star-v4 pr-star-v4-100-filled"></div>
+          <div class="pr-star-v4 pr-star-v4-0-filled"></div>
+          <div class="pr-star-v4 pr-star-v4-0-filled"></div>
+          <div class="pr-star-v4 pr-star-v4-0-filled"></div>
+          <div class="pr-star-v4 pr-star-v4-0-filled"></div>
+          <span class="pwrStarCont" style="position: relative !important;">
+            <span class="pwrStar_top" style="width:27px !important"></span>
+            <span class="pwrStar_bottom"></span>
+          </span>
+        </div>
+        """,
+    },
+]
```

Comment:
It’s a bit sad that we do not support https://books.toscrape.com/ :slightly_smiling_face: